### ISSUE TITLE
Is this for building PA applications or installing the platform?

### ISSUE THREAD

**Issue by eMPee584** [uroesch/PortableApps.comInstaller#2]
Hi Urs, just a quick question while procrastinating on our https://schulstick.org/ Förderantrag: can this be used to install the platform (in a scriptable way? like asked for in https://portableapps.com/node/57649 ..) or do you use it to build your various PA apps? Please excuse me for taking the shortcut to ask instead of figuring out myself ; )

**Comment by uroesch** [comment]
Greetings Marcel

I'm not sure I can help you with your request. Let me rephrase we are on the same page.

If I understand it correctly you want to automate the installation of the PortableApps Platform including all the apps, correct?

To the best of my knowledge and I'm not really an expert in this field you have can do as the thread mentions.
1. Install PortableApps Platform and all the apps you want all the apps some where on your hard drive e.g. `C:\PA-goldenimage`
2. To put the cherry on top I'd also add a file called `C:\PA-goldenimage\Autorun.inf` with the following content
   ```ini
   [AutoRun]
   label = Der Schulstick
   icon = Start.exe
   open = Start.exe
   ```
3. Copy everything to the root of the USB-stick. In my case I'd automate it with a script or two.

If you are looking for options to pass to the installers for silent install I have documented all the ones I could find here.
https://github.com/uroesch/PortableApps/blob/master/CommonFiles/Other/Update/PA-Upgrade.ps1#L120-L127

Hope that helps

cheers,

--urs

**Comment by uroesch** [comment]
And here is part two of the answer ;)

This repo is only used to make easier to build the apps the real thing is this repo 
- https://github.com/uroesch/PortableApps 

There is a directory called `CommonFiles\Other\Update` which holds all the build scripts. I'm honestly a lazy little sysadmin so I have automated the whole installation. The files are also found in each of the submodules as they are to the workhorses building the packages. 

For updating I have added a new INI file to each of my PortableApps called update.ini which holds the metadata to make an update as painless as possible. e.g. https://github.com/uroesch/GitPortable/blob/main/App/AppInfo/update.ini 

When a new update is announced I then run the up-release.sh script on Linux and wait until the PR closes and the release is done.  ;)

**Comment by eMPee584** [comment]
.. following up on this briefly:
> If I understand it correctly you want to automate the installation of the PortableApps Platform including all the apps, correct?

Yes, exactly.. the workflow you described will work, however, it needs a GUI to manually install the platform. For building up-to-date images regularly on a remote VM, that is less than ideal. But maybe the GUI interaction flow is reproducible enough with wine and xdotool, will have a look into that.

Thanks for the suggestion and taking the time to explain your deployment workflow. Sleek!

**Comment by uroesch** [comment]
Nope not exactly the reason there for my PortableApps collection is that I work as a consultant in many changing environments. My operating system of choice is Linux but at some clients that is not an option. Hence I started packaging opensource tools that I view as essential to do my job under Windows. What started as a manual process is now more or less automated. To be precise the download, packaging and publishing process of a new version of say [GitPortable](https://github.com/uroesch/GitPortable) is.
In order to automate the build process I forked, in a very broad sense, PortableApps.comInstaller and PortableApps.comLauncher. PortableApps.comLauncher is patched so one can generate multiple entry points. So done on the before mentioned GitPortable where there is a Git Bash, a Git CMD and a Git Gui executable. But that is only on a side note.  

The complete installation of the PortableApps I have never attempted and is not a stated goal of mine for this project. But I have quite a few Powershell snippets that might help achieve this.

As I already pointed out there are a few things you can look at: 

* I already mentioned this code where I listed all the [options you can pass on the command line](https://github.com/uroesch/PortableApps/blob/master/CommonFiles/Other/Update/PA-Upgrade.ps1#L120-L127)
* Then there is the fact that the NSIS installer is in the broader sense really an executable 7zip archive. You could use [7zip to unpack the files into the right directory as I do here](https://github.com/uroesch/PortableApps/blob/master/scripts/download-pa-dev-pkgs.ps1#L160-L171). There are a few weird looking directories left after that like the `$PLUGINSDIR` but you can remove that in a second step.

Beforehand there is the download part which [I more or less cover here](https://github.com/uroesch/PortableApps/blob/master/scripts/download-pa-dev-pkgs.ps1#L69-L110) . This is for the PortableApps.comInstaller and Launcher but you can surely use it for downloading the Platform as well. 

Sounds like a fun little project that peaked my interest I may look into it at some point. BTW do you have a project where you accept contributions?

cheers, 

--urs
